### ISSUE TITLE
[Gen2 CF-HPG v1.0 Run] Execute preregistered seed42 validation-only feasibility experiment

### ISSUE THREAD

**Issue by Irthn1311** [Irthn1311/FER2013_Graph#44]
## Objective

Execute exactly one preregistered **Generation-2 CF-HPG v1.0 seed-42 FER2013 train/validation feasibility run** from the reviewed implementation merged by PR #43.

Primary research question:

> Can the minimal convolution-free, anatomy-free CF-HPG v1.0 reach at least 0.65 validation accuracy with substantially smaller clean train-validation gaps than Generation 1?

Stretch question:

> Can the same frozen v1.0 configuration reach at least 0.70 validation accuracy without exceeding the preregistered 8 pp clean generalization-gap limits?

This issue authorizes one training/validation run only. It does not authorize test access, hyperparameter search, architecture changes, reruns driven by outcome, or CF-HPG v1.1.

## Exact authorized source

Run from exact merged `main` commit:

`562c80fe675d8527416df1b5e441df76f1f62620`

Reviewed PR #43 head:

`a84537b9cd6a6badf1a620ee7d1b20277bb30262`

Implementation base before CF-HPG:

`56a22a3200f1e0dc0100093d43f0db296c11487d`

CF-HPG model source SHA-256:

`74c29b114b37ab8b48e45eed187978213fdecb858554f4d37468042a3db4050d`

Registered model inventory:

- total/trainable parameters: `413447`
- trainable parameter variables: `64`
- Keras variables: `74`
- input: `48x48x1`
- raw non-overlapping patches: `6x6`, `64 x 36`
- fine width: `96`, exactly 2 Max-Relative blocks
- hierarchy: exact parameter-free `2x2` mean, `64 -> 16` nodes
- coarse width: `128`, exactly 2 Max-Relative blocks
- readout: global mean + global max -> `256 -> 128 -> 7`

No source/model/training configuration change is authorized in this issue.

## Dataset boundary

Dataset signature:

`fer2013_train28709_val3589_test3589`

Allowed:

- train: exactly `28,709`
- validation: exactly `3,589`

Forbidden absolutely:

- `test.csv` open/read/hash/load
- test labels/predictions/metrics
- test confusion matrix
- test smoke testing/debugging
- test-based model selection

The reviewed loader/CLI contains lexical final-split guards. Any guard failure => STOP. Do not bypass or rename a forbidden test path.

## Exact training configuration

Seed: `42`

Optimizer:

- AdamW
- initial LR `3e-4`
- weight decay `5e-4`
- global gradient norm clip `1.0`

Schedule:

- max epochs `100`
- linear warmup `5` epochs
- cosine decay afterward
- final LR `1e-6`
- no ReduceLROnPlateau

Batch size: `64`

Loss:

- categorical cross-entropy from logits
- sparse labels converted internally only for loss
- label smoothing `0.05`

Checkpoint:

- earliest strict maximum `val_accuracy`

Early stopping:

- monitor `val_loss`
- patience `15`
- min delta `0.0`
- no outcome-dependent adjustment

No MixUp, CutMix, SAM, TTA, class weighting, oversampling, alternate seed, or alternate optimizer is allowed.

## Exact train-only augmentation

Applied before patchification:

1. horizontal flip `p=0.5`
2. translation up to `+/-4` pixels x/y
3. rotation within `+/-10` degrees
4. contrast `[0.85,1.15]`
5. normalized brightness delta `[-0.10,+0.10]`
6. random erasing `p=0.25`, area `[0.02,0.10]`, aspect `[0.5,2.0]`, zero normalized fill

Geometric interpolation: bilinear.
Fill mode: `REFLECT`.

Validation and clean-train evaluation: no augmentation.

## Runtime preflight

Use Kaggle GPU runtime. Before execution record:

- exact repo HEAD
- clean tracked worktree
- Python version
- TensorFlow version
- Keras version
- visible GPU model/count
- CUDA/cuDNN identity when available
- TensorFlow global mixed-precision policy
- XLA/JIT state if available

Do not change the reviewed training source to accommodate runtime differences. Default model policy should remain whatever the reviewed CLI establishes under the clean process; do not externally monkeypatch mixed precision/XLA.

If required TensorFlow/Keras imports or the registered command cannot run cleanly, classify as technical invalid and STOP rather than modifying the model/training protocol.

## Exact authorized command

From exact checkout `562c80fe675d8527416df1b5e441df76f1f62620`:

```bash
python -m research.candidates.tf_cf_hpg.train_validation_only \
  --train-csv /kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/train.csv \
  --val-csv /kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/val.csv \
  --output-root /kaggle/working/tf_cf_hpg_v1_seed42
```

Do not add or change flags.

## One-attempt rule

This issue authorizes one actual full FER execution attempt only.

A checkout/preflight failure before the training command is launched does **not** consume the scientific attempt.

Once the training command has started, after it terminates or is runtime-censored:

- STOP
- do not rerun automatically
- do not alter hyperparameters
- do not change seed
- do not change augmentation
- do not resume from an unregistered partial checkpoint
- do not start v1.1

A second attempt requires explicit research-lead review and a written protocol amendment justified by technical failure, not performance.

## Registered endpoints

Primary endpoint:

`validation_accuracy` of the selected earliest-strict-max-val-accuracy checkpoint.

Secondary:

- validation macro-F1
- validation loss
- clean train accuracy at selected checkpoint
- clean train macro-F1 at selected checkpoint
- accuracy gap pp
- macro-F1 gap pp
- selected epoch
- epochs completed
- completion/early-stop state
- exact parameter count

Definitions:

`acc_gap_pp = 100 * (clean_train_accuracy - validation_accuracy)`

`macro_gap_pp = 100 * (clean_train_macro_f1 - validation_macro_f1)`

### `CF_HPG_V1_STRETCH_PASS`
All required:

- val accuracy `>=0.7000`
- val macro-F1 `>=0.6700`
- acc gap `<=8.0 pp`
- macro gap `<=8.0 pp`

### `CF_HPG_V1_PASS`
All required:

- val accuracy `>=0.6500`
- val macro-F1 `>=0.6200`
- acc gap `<=8.0 pp`
- macro gap `<=8.0 pp`

### `CF_HPG_V1_PERFORMANCE_WITH_GENERALIZATION_WARNING`

- val accuracy `>=0.6500`
- but either clean gap `>8.0 pp`

### `CF_HPG_V1_NEAR_TARGET`

- `0.6400 <= val accuracy < 0.6500`
- both clean gaps `<=8.0 pp`

### `CF_HPG_V1_OVERFIT`

- val accuracy `<0.6500`
- and either clean gap `>10.0 pp`

### `CF_HPG_V1_UNDERFIT`

- clean train accuracy `<0.7000`
- val accuracy `<0.6400`

### `CF_HPG_V1_INSUFFICIENT`

Any other valid result below 0.65.

These are preregistered single-seed practical labels, not statistical superiority/equivalence claims.

## Required evidence after the one run

Preserve the entire fresh output root without modifying checkpoint contents.

At minimum return/review:

- `resolved_config.json`
- `model_summary.txt`
- `checkpoints/best_val_accuracy.keras`
- `checkpoints/best_val_accuracy.metadata.json`
- `validation_only_result.json`
- training history/logs available from the execution environment
- exact command and process return code
- preflight/runtime identity
- explicit `test_access=false`

After the run terminates, create a transport ZIP of the output/evidence only for review and compute SHA-256. Do not modify model/checkpoint files before packaging.

If the training process is hard-censored before a valid final result exists, retain any partial artifacts as technical evidence only; do not infer a decision from partial metrics.

## Interpretation boundary

A valid PASS means the seed-42 candidate meets the preregistered practical performance/generalization gate. It does not establish statistical superiority or final test performance.

A STRETCH PASS would justify promoting CF-HPG as a leading graph architecture candidate for the paper, subject to confirmatory seeds and later final test closure.

A NEAR_TARGET allows at most one separately preregistered, diagnosis-driven refinement. It does not authorize a sweep.

OVERFIT/UNDERFIT determine the next hypothesis class only. Do not tune opportunistically after seeing the trajectory.

## Completion condition

This issue is complete only after:

1. exactly one authorized full train/validation attempt is executed or technically invalidated;
2. evidence is independently reviewed against this issue;
3. the registered decision is accepted or the run is declared invalid;
4. no test access occurred.

Do not start final test evaluation or CF-HPG v1.1 before independent review.

**Comment by Irthn1311** [comment]
## Independent research-lead evidence review — ACCEPTED

Reviewed the uploaded Issue #44 evidence independently against the preregistration.

### Provenance / execution
- authorized checkout: `562c80fe675d8527416df1b5e441df76f1f62620` — PASS
- clean tracked worktree — PASS
- model SHA-256: `74c29b114b37ab8b48e45eed187978213fdecb858554f4d37468042a3db4050d` — PASS
- model params: `413447` — PASS
- exact registered train/validation command — PASS
- process return code `0` — PASS
- one training attempt observed; no retry — PASS
- `test_access=false`; no test-named source/artifact observed — PASS

### Artifact integrity
Independent local review of the submitted transport archive:
- SHA-256: `dffbde44620a7e1b46c9d4536ad991cdb10238a0f560b8a3fbf9348950b9b323`
- bytes: `4,604,514`
- members: `7`
- ZIP integrity: PASS
- nested `best_val_accuracy.keras` integrity: PASS
- nested checkpoint SHA-256: `2bbe92d9ef3e465b7ca75fd2e3a4a4f5eebf6849898eec6fdc66cf4c2dd229c6`
- required result/config/summary/checkpoint metadata files present.

### Registered outcome
Selected earliest strict maximum validation-accuracy checkpoint:
- zero-based epoch `83` (epoch 84)
- validation accuracy `0.5282808581777654`
- validation macro-F1 `0.46269581824371386`
- validation loss `1.31986182778617`
- clean-train accuracy `0.5650492876798217`
- clean-train macro-F1 `0.503022788224014`
- clean-train loss `1.2362994841439476`
- accuracy gap `3.67684295020563 pp`
- macro-F1 gap `4.03269699803001 pp`

Training log independently confirms exactly 99 completed epochs, unique maximum rounded `val_accuracy=0.5283` at epoch 84, and val-loss minimum rounded `1.3199` at the same epoch; early stopping then terminates after 15 subsequent non-improving epochs.

Therefore the preregistered decision is accepted as:

`CF_HPG_V1_UNDERFIT`

This is a valid single-seed feasibility result, not a statistical claim. The small train-validation gaps are not evidence of successful generalization at useful performance because clean-train accuracy itself is only ~56.5%.

### XLA note
Preflight recorded no externally enabled XLA/JIT state, while TensorFlow later logged one internally compiled XLA cluster. This is recorded as a runtime note, not a protocol violation: source/config were unchanged and no external XLA/mixed-precision monkeypatch was applied.

### Scientific interpretation boundary
CF-HPG v1.0 is **not** a performance-success candidate. The result falsifies the current minimal `6x6 non-overlapping raw patch -> shallow patch projection -> 64-node graph -> mean hierarchy` configuration as sufficient for the >=0.65 target under the registered training protocol. It does **not** falsify the broader convolution-free/anatomy-free graph hypothesis.

Issue #44 completion conditions are satisfied. No test evaluation is authorized by this closure.
